### PR TITLE
refactor(data): replace deprecated subscribe usages

### LINKED ISSUE #3313: Replace deprecated `observable$.subscribe` usages
The `observable$.subscribe(next, error, complete)` signature is deprecated in RxJS 7 and should be replaced with `observable$.subscribe({ next, error, complete })`.

## Other information:

Deprecated `subscribe` signature is used in the following files:

- [`entity-collection-service.spec`](https://github.com/ngrx/platform/blob/master/modules/data/spec/entity-services/entity-collection-service.spec.ts)
- [`entity-cache-effects.spec`](https://github.com/ngrx/platform/blob/master/modules/data/spec/effects/entity-cache-effects.spec.ts)
- [`entity-data-module.spec`](https://github.com/ngrx/platform/blob/master/modules/data/spec/entity-data.module.spec.ts)

### PR DESCRIPTION
**Pull request by wiksien**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

Closes #3313 

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: modules/data/spec/effects/entity-cache-effects.spec.ts [hunk 1]
```diff
@@ -36,10 +36,10 @@ describe('EntityCacheEffects (normal testing)', () => {
   };
 
   function expectCompletion(completion: any, done: any) {
-    effects.saveEntities$.subscribe((result) => {
+    effects.saveEntities$.subscribe({(result) => {
       expect(result).toEqual(completion);
       done();
-    }, fail);
+    }, fail});
```
Comment (markostanimirovic) [inline modules/data/spec/effects/entity-cache-effects.spec.ts:42]:
```ts
effects.saveEntities$.subscribe({
  next: (result) => {
    expect(result).toEqual(completion);
    done();
  },
  error: fail,
});
```

File: modules/data/spec/effects/entity-cache-effects.spec.ts [hunk 2]
```diff
@@ -92,11 +92,11 @@ describe('EntityCacheEffects (normal testing)', () => {
     const action = new SaveEntities(cs, 'test/save', options);
     const cancel = new SaveEntitiesCancel(correlationId, 'Test Cancel');
 
-    effects.saveEntities$.subscribe((result) => {
+    effects.saveEntities$.subscribe({(result) => {
       expect(result instanceof SaveEntitiesSuccess).toBe(false);
       expect(result instanceof SaveEntitiesCanceled).toBe(true); // instead
       done();
-    }, done.fail);
+    }, done.fail});
```
Comment (markostanimirovic) [inline modules/data/spec/effects/entity-cache-effects.spec.ts:99]:
```ts
effects.saveEntities$.subscribe({
  next: (result) => {
    expect(result instanceof SaveEntitiesSuccess).toBe(false);
    expect(result instanceof SaveEntitiesCanceled).toBe(true); // instead
    done();
  },
  error: done.fail,
});
```

File: modules/data/spec/effects/entity-cache-effects.spec.ts [hunk 3]
```diff
@@ -108,10 +108,10 @@ describe('EntityCacheEffects (normal testing)', () => {
     const action = new SaveEntities(cs, 'test/save', options);
     const cancel = new SaveEntitiesCancel(correlationId, 'Test Cancel');
 
-    effects.saveEntities$.subscribe((result) => {
+    effects.saveEntities$.subscribe({(result) => {
       expect(result instanceof SaveEntitiesSuccess).toBe(true);
       done();
-    }, done.fail);
+    }, done.fail});
```
Comment (markostanimirovic) [inline modules/data/spec/effects/entity-cache-effects.spec.ts:114]:
```ts
effects.saveEntities$.subscribe({
  next: (result) => {
    expect(result instanceof SaveEntitiesSuccess).toBe(true);
    done();
  },
  error: done.fail,
});
```

File: modules/data/spec/effects/entity-cache-effects.spec.ts [hunk 4]
```diff
@@ -120,11 +120,11 @@ describe('EntityCacheEffects (normal testing)', () => {
 
   it('should emit SAVE_ENTITIES_SUCCESS immediately if no changes to save', (done: any) => {
     const action = new SaveEntities({ changes: [] }, 'test/save', options);
-    effects.saveEntities$.subscribe((result) => {
+    effects.saveEntities$.subscribe({(result) => {
       expect(result instanceof SaveEntitiesSuccess).toBe(true);
       expect(dataService.saveEntities).not.toHaveBeenCalled();
       done();
-    }, done.fail);
+    }, done.fail});
```
Comment (markostanimirovic) [inline modules/data/spec/effects/entity-cache-effects.spec.ts:127]:
```ts
effects.saveEntities$.subscribe({
  next: (result) => {
    expect(result instanceof SaveEntitiesSuccess).toBe(true);
    expect(dataService.saveEntities).not.toHaveBeenCalled();
    done();
  },
  error: done.fail,
});
```

File: modules/data/spec/entity-data.module.spec.ts [hunk 5]
```diff
@@ -112,7 +112,7 @@ describe('EntityDataModule', () => {
           // tap(act => console.log('test action', act)),
           skip(1) // Skip QUERY_ALL
         )
-        .subscribe((act) => actions.push(act));
+        .subscribe({(act) => actions.push(act)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-data.module.spec.ts:115]: revert this change

File: modules/data/spec/entity-data.module.spec.ts [hunk 6]
```diff
@@ -124,7 +124,7 @@ describe('EntityDataModule', () => {
       const actions: Action[] = [];
 
       // listen for actions after the next dispatched action
-      actions$.pipe(skip(1)).subscribe((act) => actions.push(act));
+      actions$.pipe(skip(1)).subscribe({(act) => actions.push(act)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-data.module.spec.ts:127]: revert this change

File: modules/data/spec/entity-data.module.spec.ts [hunk 7]
```diff
@@ -197,7 +197,7 @@ describe('EntityDataModule', () => {
         payload: data,
       };
       store.dispatch(action);
-      cacheSelector$.subscribe((cache) => {
+      cacheSelector$.subscribe({(cache) => {
```
Comment (markostanimirovic) [inline modules/data/spec/entity-data.module.spec.ts:200]:
```ts
cacheSelector$.subscribe({
  next: (cache) => {
    try {
      expect(cache.Hero.entities[1]).toEqual(data.Hero[1]);
      expect(cache.Villain.entities[30]).toEqual(data.Villain[0]);
      expect(metaReducerLog.join('|')).toContain(TEST_ACTION);
      done();
    } catch (error: any) {
      fail(error);
    }
  },
  error: fail,
});
```

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 8]
```diff
@@ -71,10 +71,10 @@ describe('EntityCollectionService', () => {
           filter((loading) => !loading),
           withLatestFrom(heroCollectionService.entities$)
         )
-        .subscribe(([loading, data]) => {
+        .subscribe({([loading, data]) => {
           expect(data).toEqual(heroes);
           done();
-        });
+        }});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:77]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 9]
```diff
@@ -83,7 +83,7 @@ describe('EntityCollectionService', () => {
       const hero2 = { id: 2, name: 'B' } as Hero;
       const heroes = [hero1, hero2];
       dataService.setResponse('getAll', heroes);
-      heroCollectionService.getAll().subscribe(expectDataToBe(heroes, done));
+      heroCollectionService.getAll().subscribe({expectDataToBe(heroes, done)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:86]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 10]
```diff
@@ -92,13 +92,13 @@ describe('EntityCollectionService', () => {
       const httpError = { error: new Error('Test Failure'), status: 501 };
       const error = makeDataServiceError('GET', httpError);
       dataService.setErrorResponse('getAll', error);
-      heroCollectionService.getAll().subscribe(expectErrorToBe(error, done));
+      heroCollectionService.getAll().subscribe({expectErrorToBe(error, done)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:95]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 11]
```diff
@@ -92,13 +92,13 @@ describe('EntityCollectionService', () => {
...
     });
 
     it('getByKey observable should emit a hero on success', (done: any) => {
       const hero = { id: 1, name: 'A' } as Hero;
       dataService.setResponse('getById', hero);
-      heroCollectionService.getByKey(1).subscribe(expectDataToBe(hero, done));
+      heroCollectionService.getByKey(1).subscribe({expectDataToBe(hero, done)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:101]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 12]
```diff
@@ -117,7 +117,7 @@ describe('EntityCollectionService', () => {
       dataService.setErrorResponse('getById', error);
       heroCollectionService
         .getByKey(42)
-        .subscribe(expectErrorToBe(error, done));
+        .subscribe({expectErrorToBe(error, done)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:120]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 13]
```diff
@@ -127,7 +127,7 @@ describe('EntityCollectionService', () => {
       dataService.setResponse('getWithQuery', heroes);
       heroCollectionService
         .getWithQuery({ name: 'foo' })
-        .subscribe(expectDataToBe(heroes, done));
+        .subscribe({expectDataToBe(heroes, done)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:130]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 14]
```diff
@@ -138,22 +138,22 @@ describe('EntityCollectionService', () => {
       dataService.setErrorResponse('getWithQuery', error);
       heroCollectionService
         .getWithQuery({ name: 'foo' })
-        .subscribe(expectErrorToBe(error, done));
+        .subscribe({expectErrorToBe(error, done)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:141]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 15]
```diff
@@ -138,22 +138,22 @@ describe('EntityCollectionService', () => {
...
     });
 
     it('load observable should emit heroes on success', (done: any) => {
       const hero1 = { id: 1, name: 'A' } as Hero;
       const hero2 = { id: 2, name: 'B' } as Hero;
       const heroes = [hero1, hero2];
       dataService.setResponse('getAll', heroes);
-      heroCollectionService.load().subscribe(expectDataToBe(heroes, done));
+      heroCollectionService.load().subscribe({expectDataToBe(heroes, done)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:149]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 16]
```diff
@@ -138,22 +138,22 @@ describe('EntityCollectionService', () => {
...
     });
 
     it('load observable should emit expected error when data service fails', (done: any) => {
       const httpError = { error: new Error('Test Failure'), status: 501 };
       const error = makeDataServiceError('GET', httpError);
       dataService.setErrorResponse('getAll', error);
-      heroCollectionService.load().subscribe(expectErrorToBe(error, done));
+      heroCollectionService.load().subscribe({expectErrorToBe(error, done)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:156]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 17]
```diff
@@ -183,14 +183,14 @@ describe('EntityCollectionService', () => {
...
-      heroCollectionService.getAll(options).subscribe(
+      heroCollectionService.getAll(options).subscribe({
         (data) => fail('should not have data but got data'),
         (error) => {
           expect(error instanceof PersistanceCanceled).toBe(true);
           expect(error.message).toBe('Test cancel');
           done();
         }
-      );
+      });
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:193]:
```ts
heroCollectionService.getAll(options).subscribe({
  next: (data) => fail('should not have data but got data'),
  error: (error) => {
    expect(error instanceof PersistanceCanceled).toBe(true);
    expect(error.message).toBe('Test cancel');
    done();
  },
});
```

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 18]
```diff
@@ -203,10 +203,10 @@ describe('EntityCollectionService', () => {
 
       const correlationId = 'CRID007';
       const options: EntityActionOptions = { correlationId };
-      heroCollectionService.getAll(options).subscribe((data) => {
+      heroCollectionService.getAll(options).subscribe({(data) => {
         expect(data).toEqual(heroes);
         done();
-      }, fail);
+      }, fail});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:209]:
```ts
heroCollectionService.getAll(options).subscribe({
  next: (data) => {
    expect(data).toEqual(heroes);
    done();
  },
  error: fail,
});
```

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 19]
```diff
@@ -221,7 +221,7 @@ describe('EntityCollectionService', () => {
       const options: EntityActionOptions = { correlationId };
       heroCollectionService
         .getAll(options)
-        .subscribe((data) => expect(data).toEqual(heroes), fail);
+        .subscribe({(data) => expect(data).toEqual(heroes), fail});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:224]:
```suggestion
        .subscribe({
          next: (data) => expect(data).toEqual(heroes),
          error: fail,
        });
```

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 20]
```diff
@@ -287,46 +287,46 @@ describe('EntityCollectionService', () => {
       dataService.setResponse('add', hero);
       heroCollectionService
         .add(hero)
-        .subscribe(expectDataToBe(hero, done, undefined, extra));
+        .subscribe({expectDataToBe(hero, done, undefined, extra)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:290]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 21]
```diff
@@ -287,46 +287,46 @@ describe('EntityCollectionService', () => {
...
     });
 
     it('add() observable should emit expected error when data service fails', (done: any) => {
       const hero = { id: 1, name: 'A' } as Hero;
       const httpError = { error: new Error('Test Failure'), status: 501 };
       const error = makeDataServiceError('PUT', httpError);
       dataService.setErrorResponse('add', error);
-      heroCollectionService.add(hero).subscribe(expectErrorToBe(error, done));
+      heroCollectionService.add(hero).subscribe({expectErrorToBe(error, done)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:298]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 22]
```diff
@@ -287,46 +287,46 @@ describe('EntityCollectionService', () => {
...
     });
 
     it('delete() should send delete for entity not in cache and return its id', (done: any) => {
       const extra = expectOptimisticSuccess(isOptimistic);
       dataService.setResponse('delete', 42);
       heroCollectionService
         .delete(42)
-        .subscribe(expectDataToBe(42, done, undefined, extra));
+        .subscribe({expectDataToBe(42, done, undefined, extra)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:306]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 23]
```diff
@@ -287,46 +287,46 @@ describe('EntityCollectionService', () => {
...
     });
 
     it('delete() should skip delete for added entity cache', (done: any) => {
       // reducedActions$Snoop();
       let wasSkipped: boolean;
-      successActions$.subscribe(
+      successActions$.subscribe({
         (act: EntityAction) => (wasSkipped = act.payload.skip === true)
-      );
+      });
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:314]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 24]
```diff
@@ -287,46 +287,46 @@ describe('EntityCollectionService', () => {
...
       const extra = () => expect(wasSkipped).toBe(true);
 
       const hero = { id: 1, name: 'A' } as Hero;
       heroCollectionService.addOneToCache(hero);
       dataService.setResponse('delete', 1);
       heroCollectionService
         .delete(1)
-        .subscribe(expectDataToBe(1, done, undefined, extra));
+        .subscribe({expectDataToBe(1, done, undefined, extra)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:322]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 25]
```diff
@@ -287,46 +287,46 @@ describe('EntityCollectionService', () => {
...
     });
 
     it('delete() observable should emit expected error when data service fails', (done: any) => {
       const httpError = { error: new Error('Test Failure'), status: 501 };
       const error = makeDataServiceError('DELETE', httpError);
       dataService.setErrorResponse('delete', error);
-      heroCollectionService.delete(42).subscribe(expectErrorToBe(error, done));
+      heroCollectionService.delete(42).subscribe({expectErrorToBe(error, done)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:329]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 26]
```diff
@@ -337,7 +337,7 @@ describe('EntityCollectionService', () => {
       dataService.setResponse('update', null); // server returns nothing after update
       heroCollectionService
         .update(update)
-        .subscribe(expectDataToBe(update, done, undefined, extra));
+        .subscribe({expectDataToBe(update, done, undefined, extra)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:340]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 27]
```diff
@@ -353,7 +353,7 @@ describe('EntityCollectionService', () => {
       dataService.setResponse('update', postUpdate); // server returns entity with server-side changes
       heroCollectionService
         .update(update)
-        .subscribe(expectDataToBe(postUpdate, done, undefined, extra));
+        .subscribe({expectDataToBe(postUpdate, done, undefined, extra)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:356]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 28]
```diff
@@ -376,7 +376,7 @@ describe('EntityCollectionService', () => {
       let responseDelay = delayMs;
       const savedHeroes: Hero[] = [];
 
-      successActions$.pipe(delay(1)).subscribe((act) => {
+      successActions$.pipe(delay(1)).subscribe({(act) => {
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:379]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 29]
```diff
@@ -400,11 +400,11 @@ describe('EntityCollectionService', () => {
       // Confirm that each add returns with its own hero
       heroCollectionService
         .add(hero1)
-        .subscribe((data) => expect(data).toEqual(hero1));
+        .subscribe({(data) => expect(data).toEqual(hero1)});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:403]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 30]
```diff
@@ -428,9 +428,9 @@ describe('EntityCollectionService', () => {
         { id: 1, name: 'A' },
       ]);
       store.dispatch(action);
-      heroCollectionService.collection$.subscribe((collection) => {
+      heroCollectionService.collection$.subscribe({(collection) => {
         expect(collection.ids).toEqual([1]);
-      });
+      }});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:433]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 31]
```diff
@@ -501,17 +501,17 @@ function entityServicesSetup() {
   function expectOptimisticSuccess(expected: boolean) {
     let wasOptimistic: boolean;
     const msg = `${expected ? 'Optimistic' : 'Pessimistic'} save `;
-    successActions$.subscribe(
+    successActions$.subscribe({
       (act: EntityAction) => (wasOptimistic = act.payload.isOptimistic === true)
-    );
+    });
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:506]: revert this change

File: modules/data/spec/entity-services/entity-collection-service.spec.ts [hunk 32]
```diff
@@ -501,17 +501,17 @@ function entityServicesSetup() {
...
     return () => expect(wasOptimistic).toBe(expected);
   }
 
   /** Snoop on reducedActions$ while debugging a test */
   function reducedActions$Snoop() {
-    reducedActions$.subscribe((act) => {
+    reducedActions$.subscribe({(act) => {
       console.log('scannedActions$', act);
-    });
+    }});
```
Comment (markostanimirovic) [inline modules/data/spec/entity-services/entity-collection-service.spec.ts:514]: revert this change

**Review by markostanimirovic — CHANGES_REQUESTED** [review]
Thanks for the PR @wiksien!

1. Run `yarn install` to install dependencies.
2. When committing changes, don't add the `--no-verify` flag to make sure the changed code is properly formatted.

**Comment by wiksien** [comment]
Sorry for causing problems with fixing this issue. I hope it is all okay now, thank you for being such a big help.

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**